controller/state.go

```diff
@@ -597,12 +627,21 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *v1
 		} else {
 			diffResult = diff.DiffResult{Modified: false, NormalizedLive: []byte("{}"), PredictedLive: []byte("{}")}
 		}
+
+		// For the case when a namespace is managed with `managedNamespaceMetadata` AND it has resource tracking
+		// enabled (e.g. someone manually adds resource tracking labels or annotations), we need to do some
+		// bookkeeping in order to ensure that it's not considered `OutOfSync` (since it does not exist in source
+		// control).
+		//
+		// This is in addition to the bookkeeping we do above (see line ~500) to prevent said namespace from being pruned.
```

Comment:
I'd argue against using `line` numbers for references.

You could try to use a code-ish path, e.g.:

> See the above `argo.GetTrackingMethod` loop that checks `app.Spec.SyncPolicy.ManagedNamespaceMetadata != nil`

Reply:
Yeah that's a good point, I didn't really like this approach myself but couldn't think of a better way. I'll amend the comment 👍